classifier/bisac.py

```diff
@@ -5,13 +5,57 @@
 import string
 from . import *
 
-# Special tokens used in matching rules.
-nonfiction = object()
-fiction = object()
-juvenile = object()
-ya = object()
-something = object()
+class CustomMatchToken(object):
+    """A custom token used in matching rules."""
+    def matches(self, subject_token):
+        """Does the given token match this one?"""
+        raise NotImplementedError()
+
+class Something(CustomMatchToken):
+    """A CustomMatchToken that will match any single token."""
+    def matches(self, subject_token):
+        return True
+
+class RE(CustomMatchToken):
+    """A CustomMatchToken that performs a regular expression search."""
+    def __init__(self, pattern):
+        self.re = re.compile(pattern, re.I)
+    
+    def matches(self, subject_token):
+        return self.re.search(subject_token)
+
+class Interchangeable(CustomMatchToken):
+    """A token that matches a list of strings."""
+    def __init__(self, *choices):
+        """All of these strings are interchangeable for matching purposes."""
+        self.choices = set(map(string.lower, choices))
+
+    def matches(self,subject_token):
+        return subject_token in self.choices
```

Comment:
I feel like this should use subject_token.lower, rather than relying on outside code doing that. Especially since the other CustomMatchToken classes aren't case-sensitive.